content/raw/wallet/token_store.md

```diff
@@ -0,0 +1,56 @@
+---
+title: "Token store"
+description: ""
+version: "1.0.0"
+date: 2022-05-19T12:03:57+00:00
+lastmod: 2022-05-18T12:03:57+00:00
+draft: false
+images: []
+menu:
+  raw:
+    parent: "wallet"
+weight: 100
+toc: true
+---
+
+# Token store Spec
+
+## Motivation
+
+This specs defines how the token store should be managed in Status applications for mainnet and test
+
+## Definitions
+
+### Token
+
+A token either implement the ERC20 specs or is native in the network.
+
+For example:
+* SNT is implementing the ERC20 specs
+* ETH is the native chain token of Ethereum mainnet
+
+## Functional Requirements
+
+### Token Identifier
+
+The way to identify a token is via the Symbol
+
+Example:
+    * SNT and DAI are 2 differents token
+
+### Token balance
+
+The balance of a token MUST be the sum of all enabled chain balance for the specific token
+Example:
+    * Account A have 10 SNT on mainnet and 10 SNT on arbitrum, the balance of this token will then be 20
+
+Native token and ERC20 are not different
+
+### Testnet Token
+
+Testnet token are not different in behaviour than mainnet token.
+The name of a token in mainnet MUST be named the same way in testnet
+
+Example:
+    * SNT MUST be named SNT on all chain, test or not.
+    * ETH MUST be named ETH on all chain, test or not.
```

Comment:
Everything looks good to me :-)  However there is one thing I would add:

What should happen if somebody unofficially deploys a bridge to a chain.  For example, on Optimism there is an official bridge contract that bridges DAI to Optimism.  DAI on Optimism that's minted from the official bridge should be called DAI.  But there is nothing stopping someone from deploying their own unofficial bridge contract that also bridges DAI to Optimism.  In this case, DAI from the unofficial bridge contract on Optomism should not be called DAI.  Note this doesn't impact the bridges we are evaluating for integration, because all these bridges are actually an additional bridge layer built on top of the official bridge contracts to allow instant bridging via liquidity pools (there are actually two categories of 'bridges' - the bottom level bridges that look after the actual bridging of tokens and the deployment of the token contracts on the L2, and upper level bridges that utilize tokens from contracts already deployed to an L2 and offer fast bridging services via liquidity pool type mechanisms.

Reply:
@alaibe ^

Reply:
@John-44 I don t understand why a bridge should influence the name of the token, DAI will always be DAI whatever the bridge no?

e.g what stopped me from creating a bridge between DAI optimism and arbitrum, they would still be DAI

Except if you mean that someone deployed an ERC20 contract and decided to named it DAI while not being DAI?
In that case, yes I agree, this is why we need to be in control of the token in different chain and if a bridge decide to name something DAI that is not DAI, we will change its name in our list of token